libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -457,6 +454,7 @@ private void startNavigation(DirectionsRoute route) {
       navigation.startNavigation(route);
       voiceInstructionsToAnnounce = 0;
       voiceInstructionCache.preCache(route);
+      navigation.addProgressChangeListener(mapOfflineManager);
```

Comment:
another potential place for adding nav progress change listener is `private void addNavigationListeners()` on line 402. it would be cleaner to add all the navigation listeners in one method but then your initialization logic has to change. One viable option would be to add listeners with every `start` call and cleanup with every `end` call. Another option would be to change initialization sequence in `initialize` method but then you'd have to ensure navigation listeners are added in the very end when all other managers are initialized and valid. i'd still highly recommend to check if every listener ref is valid before registering them as we have few optional ones.

Reply:
`mapOfflineManager` is an optional dependency, if user end up not providing map options this reference will never be initialized and as a result we shall be passing `null` to the `addProgressChangeListener` method, which in turn can result in crash or not. have we tested this scenario?

Reply:
@olegzil fyi your instrumented tests are failing we'd have to fix them